pkgs/ffigen/lib/src/generator.dart

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:cli_util/cli_logging.dart' show Ansi;
+import 'package:logging/logging.dart';
+
+import 'config_provider.dart' show Config;
+import 'header_parser.dart' show parse;
+
+final _logger = Logger('ffigen.generate');
+final _ansi = Ansi(Ansi.terminalSupportsAnsi);
+bool _loggerSetup = false;
+
+/// Runs the entire generation pipeline for the given config.
+void generate(Config config) {
+  if (!_loggerSetup) {
+    // Set up the logger if it hasn't already been set up.
+    setupLogger(null);
+  }
+
+  // Parse the bindings according to config object provided.
+  final library = parse(config);
+
+  // Generate files for the parsed bindings.
+  final gen = File(config.output);
+  library.generateFile(gen, format: config.formatOutput);
+  _logger
+      .info(successPen('Finished, Bindings generated in ${gen.absolute.path}'));
+
+  final objCGen = File(config.outputObjC);
+  if (library.generateObjCFile(objCGen)) {
+    _logger.info(successPen('Finished, Objective C bindings generated '
+        'in ${objCGen.absolute.path}'));
+  }
+
+  if (config.symbolFile != null) {
+    final symbolFileGen = File(config.symbolFile!.output);
+    library.generateSymbolOutputFile(
+        symbolFileGen, config.symbolFile!.importPath);
+    _logger.info(successPen(
+        'Finished, Symbol Output generated in ${symbolFileGen.absolute.path}'));
+  }
+}
+
+/// Sets up the logging level and printing.
+void setupLogger(Level? level) {
```

Comment:
Make a class `FfiGen` that takes `Level` (or a `Logger`) in the constructor, and make `generate` an instance method on it.

Reply:
Done